FreeRTOS/Test/CMock/timers/timers_utest.c

```diff
@@ -81,31 +89,34 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
-
 static void _xCallback_Test( TimerHandle_t xTimer )
-{}
+{
+}
 
 /**
  * @brief xTimerCreate happy path
- * 
+ *
  */
 void test_xTimerCreate_Success( void )
 {
-    uint32_t ulID = 0; 
+    uint32_t ulID = 0;
     TimerHandle_t xTimer = NULL;
 
     vListInitialise_Ignore();
-    xQueueGenericCreateStatic_IgnoreAndReturn( (QueueHandle_t)1 );
+    xQueueGenericCreateStatic_IgnoreAndReturn( ( QueueHandle_t ) 1 );
     vQueueAddToRegistry_Ignore();
     vListInitialiseItem_Ignore();
 
     xTimer = xTimerCreate( "ut-timer",
-                            pdMS_TO_TICKS(1000),
-                            pdTRUE,
-                            &ulID,
-                            _xCallback_Test );
+                           pdMS_TO_TICKS( 1000 ),
+                           pdTRUE,
+                           &ulID,
+                           _xCallback_Test );
 
     TEST_ASSERT_NOT_EQUAL( NULL, xTimer );
+
+    /* HACK: Free the timer directly */
```

Comment:
Is there a proper solution instead of hack?

Reply:
Currently timers_utest.c is just a stub.

timers.c uses a command queue mechanism to free timers in the timer service task. 

With the current environment, there is no way to exercise that code since it resides in a static function (prvProcessReceivedCommands) which is only called from within a for( ; ; ) loop in prvTimerTask.

A short term option could be to remove this test case until the correct structure in place to exercise prvProcessReceivedCommands.

Long term, we should either 
(1) Decide how to best access the static functions in timers.c for unit testing purposes OR
(2) Remove the tasks.h mock for this test and run prvTimerTask in some kind of simulated environment.